packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -285,6 +292,35 @@ const SigninContainer = ({
         }
       );
 
+      // Check recovery key status if signin was successful, user is on sync Desktop
+      // and they didn't click "Do it later"; this affects navigation.
+      if (
+        'data' in result &&
+        result.data &&
+        options.service === 'sync'
+        // && localStorage.getItem('TODO')
```

Comment:
Currently tweaking this. Instead of having this check in `getSyncNavigate` too, I'm just going to do it here only and rename `result.data.recoveryKeyExists`.